### PR TITLE
Improved types for generated service arguments

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
Improved the parameter types for some generated service arguments. specifically:
date => DateOnly
time => TimeOnly
datetime => DateTime
rgb_color => IReadOnlyCollection<int>,
number without 'step' is now double instead of long

This will allow for eg:
Light.Attic.TurnOn(transition: 0.5, rgbColor: [255,128,10]);

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] The code compiles without warnings (code quality check)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration are added/changed:

- [ ] Documentation added/updated for http://netdaemon.xtz

[Docs repository]: https://github.com/net-daemon/docs
[Developer documentation]: https://netdaemon.xyz/docs/developer/

### CODE REVIEW

File: src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ServiceMetaDataParserTest.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void TestSomeBasicServicesCanBeParsed()
     [Fact]
     public void TestServicesWithAdvancedFieldsCanBeParsed()
     {
-        var sample = File.ReadAllText(@"CodeGenerator/ServiceMetaDataSamples/Lights.json");
+        var sample = File.ReadAllText(@"CodeGenerator/ServiceMetaDataSamples/light.json");
```
Comment (Barsonax) [inline src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ServiceMetaDataParserTest.cs:49]: Would a raw string literal be a better way to store the json string? Then you don't have to do any file IO in the tests.

Comment (FrankBakkerNl) [reply src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ServiceMetaDataParserTest.cs:49]: For these very long ones it seems to me it clutters the test to put them inline. That is arbitrary of course

Comment (Barsonax) [reply src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ServiceMetaDataParserTest.cs:49]: You could put them in a separate class as constants (especially when these are used in multiple tests)

**Review by helto4real — APPROVED** [review]
Looks good to me, I think this breaking change is manageable and adds enough value

**Pull request merged**